### DISCUSSION TITLE
Unable to run in exclusive fullscreen

### DISCUSSION THREAD

**Discussion by Ladwick** [ppy/osu#21651] [Q&A] [answered]
This is my first time starting a discussion, so I'm sorry if I miss something in advance!

**THE PROBLEM**

This started to happen to me quite a few updates ago, not sure about which one exactly, probably around july.

I remember looking for a way to fix this back in the day when it started happening, I'm not sure which update started this problem (but probably around halfyear), the thing is that since [PPY said himself there was a false positive with the exclusive fullscreen thing on lazer](https://github.com/ppy/osu/discussions/20232), I didn't paid much attention to it, but even when they fixed the false positive problem in lazer ver. [2022.1008.0](https://osu.ppy.sh/home/changelog/lazer/2022.1008.0) the problem persisted in my game, so I think it's not a false positive in the end.

![image](https://user-images.githubusercontent.com/37979533/207937248-41e8ae20-964a-4049-8635-fa289287a29a.png) lazer ver. [2022.1208.0](https://osu.ppy.sh/home/changelog/lazer/2022.1208.0)

My system runs both a integrated **Intel UHD Graphics** and a discrete **RTX 2070** GPU, so I'm not sure if this has something to do with the issue, I can't neither run the exclusive fullscreen in stable, so probably is a driver issue, I have all my drivers up to date but nothing seems to work.

I also found [this issue](https://github.com/ppy/osu-stable-issues/issues/412) on the stable branch, they say that using the [26.20.100.7000](https://downloadcenter.intel.com/download/28991/Intel-Graphics-Windows-10-DCH-Drivers) version of Intel drivers fix the problem, but I can't find those drivers on the Intel website so I haven't been able to try that myself :/

**Comment by Doksha** [comment]
Have you tried using the NVIDIA control panel to select your RTX 2070 as the selected card for lazer? (manage 3D settings > Program settings> select add and then find lazer > select your nvidia card from the "select the preferred graphics processor for this program" drop down. Then go to Manage Display Mode and select your NVIDIA card.)

this has worked for me before, then I just kinda started ignoring that because the game felt responsive *enough* and it was to tedious to get it to work (though the issue still happens)

Its also entirely possible that peppy thinks the issue was fixed and either it simply wansn't, or the fix wasnt actually implemented (no fault on him, it just sometimes happens)

**Reply by Ladwick** [reply]
> Have you tried using the NVIDIA control panel to select your RTX 2070 as the selected card for lazer? (manage 3D settings > Program settings> select add and then find lazer > select your nvidia card from the "select the preferred graphics processor for this program" drop down. Then go to Manage Display Mode and select your NVIDIA card.)
> 
> this has worked for me before, then I just kinda started ignoring that because the game felt responsive _enough_ and it was to tedious to get it to work (though the issue still happens)
> 
> Its also entirely possible that peppy thinks the issue was fixed and either it simply wansn't, or the fix wasnt actually implemented (no fault on him, it just sometimes happens)

Yeah, it's not his fault, I just thought that giving that piece of information was important, he already does a lot for the community.

Regarding the NVIDIA control panel, yeah, I tried that already, I even did that same thing on the windows graphic elements settings just to be sure, but it didn't worked neither. At this point I really think that going through all of these problems to fix it and reducing probably no more than 1ms is not worth it, but maybe more people has a similar problem to me, and maybe they would like to fix the problem, if I'm able to help fixing it, it would be great :)

**Comment by peppy** [comment, marked as answer]
We're moving away from OpenGL at full speed, which should fix this issue once and for all. So please wait for that.

**Reply by Hasster1** [reply]
This is probably spam, but i wanted to ask: To which backend are you moving to?

**Reply by frenzibyte** [reply]
There'll be support for DirectX, Vulkan, and Metal.
